gibbon-compiler/src/Gibbon/Passes/FreshConstructors.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts  #-}
+
+module Gibbon.Passes.FreshConstructors (freshConstructors) where 
```

Comment:
Please, add haddock-comments for every of the three passes you add: module-level and the main-function level. For inspiration, see `Fllatten.hs`:

- module-level: https://github.com/iu-parfunc/gibbon/blob/efd6b60cb6b5f56082c39a8ab351767fe9d5d742/gibbon-compiler/src/Gibbon/Passes/Flatten.hs#L6-L7
- main-function level: https://github.com/iu-parfunc/gibbon/blob/efd6b60cb6b5f56082c39a8ab351767fe9d5d742/gibbon-compiler/src/Gibbon/Passes/Flatten.hs#L24-L31